cmd/operator-sdk/up/local.go

```diff
@@ -177,17 +177,12 @@ func setupOperatorEnv() error {
 }
 
 func buildLocal(outputBinName string) error {
-	args := []string{"build", "-o", outputBinName}
+	var args []string
 	if ldFlags != "" {
-		args = append(args, "-ldflags", ldFlags)
+		args = []string{"-ldflags", ldFlags}
 	}
-	args = append(args, filepath.Join(scaffold.ManagerDir, scaffold.CmdFile))
-
-	bc := exec.Command("go", args...)
-	if err := projutil.ExecCmd(bc); err != nil {
-		return err
-	}
-	return nil
+	managerDir := filepath.Join(scaffold.ManagerDir, scaffold.CmdFile)
+	return projutil.GoBuild(outputBinName, managerDir, args...)
```

Comment:
This won't work for non-Linux OSes because `GoBuild()` uses `"GOOS=linux", "GOARCH=amd64", "CGO_ENABLED=0"`. `up local` should build the binary for the current environment.

Another place we need go builds for the current environment is: https://github.com/operator-framework/operator-sdk/blob/52ad33a4bda0f78bdc8acdc49586a670fc755de5/cmd/operator-sdk/internal/genutil/genutil.go#L43

Should we have separate build functions or add an environment variables argument to `GoBuild()`?

Reply:
Good point. Adding an environment variable arg is easiest, and I'll set the default if env variables are not set to the above. I'll create a `GoBuildOptions` type to simplify `GoBuild()` args.